packages/typespec-python/src/emitter.ts

```diff
@@ -72,18 +75,26 @@ export async function $onEmit(context: EmitContext<PythonEmitterOptions>) {
     const yamlMap = emitCodeModel(sdkContext);
     addDefaultOptions(sdkContext);
     const yamlPath = await saveCodeModelAsYaml("typespec-python-yaml-map", yamlMap);
-    const commandArgs = [
-        `${root}/scripts/run-python3.cjs`,
-        `${root}/scripts/run_tsp.py`,
-        `--output-folder=${outputDir}`,
-        `--cadl-file=${yamlPath}`,
-    ];
+    const platform = os.platform();
+    var venv_path = "unknown";
+    switch(platform){
+        case "linux":
+            venv_path = `${root}/venv/bin/python`;
+        case "darwin":
+            venv_path = `${root}/venv/bin/python`;
+        case "win32":
+            venv_path = `${root}\\venv\\Scripts\\python.exe`;
+    }
+    if (!fs.existsSync(venv_path)){
+        throw new Error("Virtual environment doesn't exist.")
+    }
+    const commandArgs = `${venv_path} ${root}/scripts/run_tsp.py --output-folder=${outputDir} --cadl-file=${yamlPath}`
```

Comment:
i think it's easier if we keep dealing with `commandArgs` as a list of arguments, like how we're currently dealing with it, then joining at the end